sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionContainer.java

```diff
@@ -223,7 +224,7 @@ private void setSessionToken(ResourceId resourceId, String collectionName, Strin
         if (!isKnownCollection) {
             this.writeLock.lock();
             try {
-                if (collectionName != null && resourceId.getUniqueDocumentCollectionId() != 0) {
+                if (resourceId.getUniqueDocumentCollectionId() != 0) {
```

Comment:
why are we making these changes? removing these sanity checks makes me a bit uncomfortable.

Reply:
Had `collectionName` is null then there will be a failure even before we reach the line 227.

Below code has this `this.collectionNameToCollectionResourceId.containsKey(collectionName) ` which expect collectionName to be not-null, otherwise it is NPE here it self.

```java
try {
    isKnownCollection = this.collectionNameToCollectionResourceId.containsKey(collectionName) &&
            this.collectionResourceIdToCollectionName.containsKey(resourceId.getUniqueDocumentCollectionId()) &&
            this.collectionNameToCollectionResourceId.get(collectionName) == resourceId.getUniqueDocumentCollectionId() &&
            this.collectionResourceIdToCollectionName.get(resourceId.getUniqueDocumentCollectionId()).equals(collectionName);
    if (isKnownCollection) {
        this.addSessionToken(resourceId, partitionKeyRangeId, parsedSessionToken);
    }
} finally {
    this.readLock.unlock();
}

if (!isKnownCollection) {
    this.writeLock.lock();
    try {
        if (resourceId.getUniqueDocumentCollectionId() != 0) {
            this.collectionNameToCollectionResourceId.compute(collectionName, (k, v) -> resourceId.getUniqueDocumentCollectionId());
            this.collectionResourceIdToCollectionName.compute(resourceId.getUniqueDocumentCollectionId(), (k, v) -> collectionName);
        }
        addSessionToken(resourceId, partitionKeyRangeId, parsedSessionToken);
    } finally {
        this.writeLock.unlock();
    }
}
```

Reply:
it seems the entire method expects a non-null collectionName, how about we add a guard null-check in this method?

Reply:
Yes, that would be great. Let's add that.

Reply:
@anuchandy can we roll back the change on SessionContainer and add a spotbug override on this one?

Reply:
Ok, I will rollback this change + add supression

Reply:
done in the lastest [commit](https://github.com/Azure/azure-sdk-for-java/pull/9111/commits/b972e8c088bb80842e7fa0633740f49128f1991b).